### ISSUE TITLE
ComboBox raises selection property change even though it does not support the selection provider.

### ISSUE THREAD

**Issue by Tanya-Solyanik** [dotnet/winforms#7764]
ComboBox AccessibleObject [raises](https://github.com/dotnet/winforms/blob/7f4adcd84d61704d0ed5164b43b94e4667831225/src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ComboBoxAccessibleObject.cs#L273) [ElementSelected property change event](https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-implementingselectionitem), however this control does not support the [SelectionProvider](https://docs.microsoft.com/en-us/windows/win32/api/uiautomationcore/nn-uiautomationcore-iselectionitemprovider). 
[Docs](https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-implementingselectionitem) specify this event/provider combination as required. 

*Result:*
Accessibility clients don't listen to events associated with this provider. Specifically, NVDA does not listen to it, resulting in compliance issue in VS property pages.

We can't remove this event because other clients are likely to rely on it. However, we can implement the missing pattern.

The same issue is present on .NET Framework 481,

*Note:*
This bug was discovered while testing VS property pages under NVDA. Because winforms ComboBox does not report Selection provider as supported, NVDA is not listening to selection change event. From @rperez030
```
I had a chat with an NVDA developer yesterday, and he pointed me to the place in the source code where they are registering the elementSelected event. He believes it may not require any work on NVDA's site for it to work once the control exposes the proper patterns.
```

**Comment by dmitrii-drobotov** [comment]
From ComboBox [docs](https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-supportcomboboxcontroltype#required-control-patterns): 
> ISelectionProvider — Displays the current selection in the combo box. Support for the [Selection](https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-implementingselection) control pattern **is delegated to the list box beneath the combo box**, but may not always be feasible.

And the child list box actually [supports](https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ComboBoxChildListUiaProvider.cs#L214) the selection pattern.
![Screenshot 2022-09-15 191625](https://user-images.githubusercontent.com/102954094/190469080-f8713944-1aef-4c50-b2a9-9efa420ecdb8.png)

So it looks like we're doing it correctly on our side.

**Comment by Tanya-Solyanik** [comment]
Makes sense, thank you for the investigation.
